gluten-substrait/src/main/scala/org/apache/spark/sql/execution/GlutenAutoAdjustStageResourceProfile.scala

```diff
@@ -88,6 +88,11 @@ case class GlutenAutoAdjustStageResourceProfile(glutenConf: GlutenConfig, spark:
       val newExecutorMemory =
         new ExecutorResourceRequest(ResourceProfile.MEMORY, newMemoryAmount.toLong)
       executorResource.put(ResourceProfile.MEMORY, newExecutorMemory)
+
+      val newExecutorOffheap =
+        new ExecutorResourceRequest(ResourceProfile.OFFHEAP_MEM, offheapRequest.get.amount / 10)
```

Comment:
@zjuwangg, could you take a look? This is just an empirical setting. And I am not sure which setting is better.

Reply:
LGTM in whole stage fallback case!

Reply:
This would work only for Spark 3.5.4 or higher as https://github.com/apache/spark/pull/48963

Reply:
@PHILO-HE Should we consider making this setting configurable?

Reply:
@JkSelf, let's first use this empirical fixed setting and wait for feedback from user side. In theory, user don't need to configure too much offheap memory in stage fallback case. Thanks!

Reply:
@PHILO-HE 
Thanks for improving on this!
just to confirm, I can see the code logic is to reduce the offheap memory size to 1/10, but on-heap memory size is not increased, is this intended? 

Thanks, -yuan